packages/react-components/theme-designer/src/ThemeDesigner.tsx

```diff
@@ -40,23 +41,68 @@ const useStyles = makeStyles({
 export const ThemeDesigner: React.FC<ThemeDesignerProps> = props => {
   const styles = useStyles();
 
+  const [theme, setTheme] = React.useState<string>('Custom');
   const [keyColor, setKeyColor] = React.useState<string>('#006bc7');
+  const [hueTorsion, setHueTorsion] = React.useState<number>(0);
+  const [darkCp, setDarkCp] = React.useState<number>(2 / 3);
+  const [lightCp, setLightCp] = React.useState<number>(1 / 3);
+
   const debouncedKeyColor = useDebounce(keyColor, 400);
 
-  const themes = React.useMemo(() => {
-    const brand = getBrandTokensFromPalette(debouncedKeyColor);
+  const customTheme = React.useMemo(() => {
+    const brand = getBrandTokensFromPalette(debouncedKeyColor, {
+      hueTorsion: hueTorsion,
+      darkCp: darkCp,
+      lightCp: lightCp,
+    });
     return {
       brand: brand,
       lightTheme: createLightTheme(brand),
       darkTheme: createDarkTheme(brand),
     };
-  }, [debouncedKeyColor]);
+  }, [hueTorsion, darkCp, lightCp, debouncedKeyColor]);
+
+  const themes = React.useMemo(() => {
```

Comment:
This is another area for possible cleanup. This memo seems pretty heavy for an if/else statement. But again, doing an all up refactor of state will help this.